### ISSUE TITLE
It is not possible to copy text to clipboard from error message on development page.

### ISSUE THREAD

**Issue by maxim-boligatov** [aspnet/Diagnostics#319]
_From @maxim-boligatov on July 20, 2016 7:51_

When exception occurs development page is displayed in browser.  
Sometimes it significantly increases productivity when it is possible to copy error text or sample message from error page.  
For example for the following example I can just find RenderSection line in source:

```
         <!-- Styles -->
         @this.RenderSection("styles", required: false)
        </head>
        <body>
            @RenderBody()
        </body>
        </html>

```

But due to interactivity added to development page (all mouse events are handled) I cannot copy text to clipboard easy. So I have to extract it from HTML using firebug...
How can it be fixed/improved? What do you think?

_Copied from original issue: aspnet/Mvc#5048_

**Comment by maxim-boligatov** [comment]
_From @pranavkm on July 20, 2016 13:46_

Could you file this bug in the [Diagnostics](https://github.com/aspnet/Diagnostics) repo?

**Comment by Eilon** [comment]
I thought at the bottom of the Diagnostics page there's a "Show raw details" link that makes it easy to copy & paste?

**Comment by Eilon** [comment]
Indeed, at the bottom of the developer exception page there's a "Show raw exception details" link that shows the plaintext exception message and stack trace.

If you click that link, you get text that's easy to copy and paste:

![raw-exception-details](https://cloud.githubusercontent.com/assets/202643/17000004/86b61340-4e75-11e6-9bb6-3d7f68471e01.png)

**Comment by pranavkm** [comment]
I thought @maxim-boligatov was referring to the actual source code (i.e. the code from the cs \ cshtml file) rather than the stack trace?

**Comment by Eilon** [comment]
Oh the source code being copyable? I'm not sure why you'd want to copy the source code... it's clearly not good source code if it threw an exception 😄

**Comment by maxim-boligatov** [comment]
It is good explained above.. a search of copied string it is the fastest way to find the line which was a reason of an error (we cannot click on error and open VS with related source code... at least for now and without browser plugins).  
I think you agree that each error message will sound like "This specific feature is failed during this specific action". In case of localized resources it will be still some unique piece of source code. So I just need to search it in solution... Analyzing of stack trace will be much more slower to identify code location.

In other words due to new interactivity "features" exception page does not allow text copying.
And this interactivity is really strange decision... Do you need more space for information? Put it into 2 columns... No professional developer works on monitor with resolution less than FullHD...
I hope in the future MS will not come to the market with Visual Studio Nokia 3310 Edition...
Let's avoid unnecessary clicks... like Show raw bla-bla... scrolling is faster... much faster from point of view of usability...

**Comment by maxim-boligatov** [comment]
Why is it closed?

**Comment by Eilon** [comment]
Ok we took a further look at this and I agree that there are some usability problems on this error page.

Some things we should fix:
1. The way that the CSS and clickability of this page are done needs improvement. Instead of entire rows being marked as "clickable" there should be a little expand/collapse button that is clickable.
2. Get rid of the row highlight-on-hover stuff. It's irritating.
3. Make the "Show raw exception details" link be more usable by making it look nicer, and also support expand/collapse (not just expand)

**Comment by Eilon** [comment]
@kichalla this is the issue I discussed with you yesterday.
